hydra/_internal/config_loader_impl.py

```diff
@@ -522,6 +522,21 @@ def _compose_config_from_defaults_list(
                         f"In '{default.config_path}': Validation error while composing config:\n{e}"
                     ).with_traceback(sys.exc_info()[2])
 
+        # # remove remaining defaults lists from all nodes.
+        def strip_defaults(cfg: Any) -> None:
+            if isinstance(cfg, DictConfig):
```

Comment:
It took a while to convince myself that an `if isinstance(cfg, ListConfig)` branch is not needed here.

If a list contains a missing value, you cannot use a higher-level defaults list to merge into that missing value. This is because list-merge results in the whole list being overwritten (rather than in the entries of the list being modified).

I discovered bug https://github.com/omry/omegaconf/issues/788 while working on this.